demos/classification_benchmark_demo/cpp/main.cpp

```diff
@@ -26,90 +26,90 @@
+DEFINE_string(labels, "", labels_msg);
+
+constexpr char layout_msg[] = "specify inputs layouts."
+    "Ex. \"[NCHW]\" or \"input1[NCHW],input2[NC]\" in case of more than one input";
+DEFINE_string(layout, "", layout_msg);
+
+constexpr char nireq_msg[] = "number of infer requests";
+DEFINE_uint32(nireq, 0, nireq_msg);
+
+constexpr char nstreams_msg[] = "specify count of streams";
+DEFINE_string(nstreams, "", nstreams_msg);
+
+constexpr char nt_msg[] = "number of top results. Must be >= 1. Default value is 5";
+DEFINE_uint32(nt, 5, nt_msg);
+
+constexpr char nthreads_msg[] = "specify count of threads";
+DEFINE_uint32(nthreads, 0, nthreads_msg);
+
+constexpr char res_msg[] = "set image grid resolution in format WxH. Default value is 1280x720";
+DEFINE_string(res, "1280x720", res_msg);
+
+constexpr char show_msg[] = "disable showing of processed images";
+DEFINE_bool(show, true, show_msg);
+
+constexpr char time_msg[] = "time in seconds to execute program. Default is -1 (infinite time)";
+DEFINE_uint32(time, std::numeric_limits<gflags::uint32>::max(), time_msg);
+
+constexpr char u_msg[] = "list of monitors to show initially";
+DEFINE_string(u, "", u_msg);
+
+void parse(int argc, char *argv[]) {
+    gflags::ParseCommandLineFlags(&argc, &argv, false);
+    if (FLAGS_h || 1 == argc) {
+        std::cout <<   "\t[ -h]                   " << h_msg
+                << "\n\t  -i <INPUT>            " << i_msg
+                << "\n\t  -m <MODEL FILE>       " << m_msg
+                << "\n\t[--auto_resize]         " << res_msg
+                << "\n\t[ -d <DEVICE>]          " << d_msg
+                << "\n\t[--gt \"<path>\"]       " << gt_msg
+                << "\n\t[--labels \"<path>\"]   " << labels_msg
+                << "\n\t[--layout \"<string>\"] " << layout_msg
+                << "\n\t[--nireq <NUMBER>]      " << nireq_msg
+                << "\n\t[--nstreams <NUMBER>]   " << nstreams_msg
+                << "\n\t[--nt <NUMBER>]         " << nt_msg
+                << "\n\t[--nthreads <NUMBER>]   " << nthreads_msg
+                << "\n\t[--res \"<WxH>\"]       " << res_msg
+                << "\n\t[--show] ([--noshow])   " << show_msg
+                << "\n\t[--time <NUMBER>]          " << time_msg
+                << "\n\t[ -u]                   " << u_msg
+                << "\n\tKey bindings:"
+                    "\n\t\tQ, q, Esc - Quit"
+                    "\n\t\tR, r, SpaceBar - Restart testing"
+                    "\n\t\tC - average CPU load, D - load distrobution over cores, M - memory usage, H - hide\n";
         showAvailableDevices();
         return false;
-    }
-
-    if (FLAGS_i.empty()) {
+    } if (FLAGS_i.empty()) {
         throw std::logic_error("Parameter -i is not set");
-    }
-
-    if (FLAGS_m.empty()) {
+    } if (FLAGS_m.empty()) {
         throw std::logic_error("Parameter -m is not set");
-    }
-
-    if (FLAGS_labels.empty()) {
+    } if (FLAGS_labels.empty()) {
```

Comment:
In help message you specified labels as optional, but in fact it's required. Maybe it's time to make labels optional in this demo, because providing labels arg could be a bit tiresome sometimes:)

Reply:
I am not sure that this decision is a part of the task, so for now I just return its status to 'required'.
Maybe @Wovchena has some ideas about that.

Reply:
There is no reason to run classification without knowing labels. If it was detection, there would be a way to represent results with just bboxes. But for classification labels are essential

Reply:
There might be reasons when you don't need classification labels, maybe you just want to check something in demo and actual result don't bother you. Demo could run without labels successfully, just in output instead of class names would be indexes. For some it would be enough. And for python demo we allow running without labels btw.

Reply:
> maybe you just want to check something

run benchmark_app

> in output instead of class names would be indexes

What would one do with them?

Reply:
> What would one do with them?

Don't know, but I wouldn't deny that one might want to do it, and I see no reason to forbid it. And it's just more convenient to be able to run the demo without providing one more argument.

Reply:
Ok

Reply:
This is still applicable.